components/certificate-mgt/org.wso2.carbon.identity.certificate.management.core/src/main/java/org/wso2/carbon/identity/certificate/management/core/CertificateManagementServiceImpl.java

```diff
@@ -0,0 +1,174 @@
+    private static final CertificateManagementService INSTANCE = new CertificateManagementServiceImpl();
+    private final CertificateManagementDAOImpl certificateDAO;
+
+    private CertificateManagementServiceImpl() {
+
+        certificateDAO = new CertificateManagementDAOImpl();
+    }
+
+    public static CertificateManagementService getInstance() {
+
+        return INSTANCE;
+    }
+
+    /**
+     * Add a certificate.
+     *
+     * @param certificate  Certificate information.
+     * @param tenantDomain Tenant domain.
+     * @return Certificate ID.
+     * @throws CertificateMgtException If an error occurs while adding the certificate.
+     */
+    @Override
+    public String addCertificate(Certificate certificate, String tenantDomain) throws CertificateMgtException {
+
+        LOG.debug("Adding certificate with name: " + certificate.getName());
+        doPreAddCertificateValidations(certificate);
+        String generatedCertificateId = UUID.randomUUID().toString();
+        certificateDAO.addCertificate(generatedCertificateId, certificate,
+                IdentityTenantUtil.getTenantId(tenantDomain));
+
+        return generatedCertificateId;
+    }
+
+    /**
+     * Get certificate information with given id.
+     *
+     * @param certificateId Certificate ID.
+     * @param tenantDomain  Tenant domain.
+     * @return Certificate information.
+     * @throws CertificateMgtException If an error occurs while getting the certificate.
+     */
+    @Override
+    public Certificate getCertificate(String certificateId, String tenantDomain) throws CertificateMgtException {
+
+        LOG.debug("Retrieving certificate with id: " + certificateId);
+        Certificate certificate = certificateDAO.getCertificate(certificateId,
+                IdentityTenantUtil.getTenantId(tenantDomain));
+        if (certificate == null) {
+            LOG.debug("No certificate found for the id: " + certificateId);
+            CertificateMgtExceptionHandler.throwClientException(CertificateMgtErrors.ERROR_CERTIFICATE_DOES_NOT_EXIST,
+                    certificateId);
+        }
+
+        return certificate;
+    }
+
+    /**
+     * Update a certificate with given id.
+     * Only the non-null and non-empty fields in the provided certificate will be updated.
+     * Null or empty fields will be ignored.
+     *
+     * @param certificateId Certificate ID.
+     * @param certificate   Certificate content.
+     * @param tenantDomain  Tenant domain.
+     * @throws CertificateMgtException If an error occurs while updating the certificate.
+     */
+    @Override
+    public void updateCertificate(String certificateId, Certificate certificate, String tenantDomain)
+            throws CertificateMgtException {
+
+        LOG.debug("Updating certificate with id: " + certificateId);
+        checkIfCertificateExists(certificateId, tenantDomain);
+        if (certificate.getName() != null && certificate.getCertificate() != null) {
```

Comment:
Use StringUtils.isNotBlank()

Reply:
This logic is removed hence we dont need to modify this